d365fo.tools/functions/invoke-d365bestpractice.ps1

```diff
@@ -0,0 +1,87 @@
+﻿
+<#
+    .SYNOPSIS
+        Runs the Best Practice checks
+        
+    .DESCRIPTION
+        A cmdlet that wraps some of the cumbersome work into a streamlined process
+        
+    .PARAMETER BinDir
+        The path to the bin directory for the environment
+        
+        Default path is the same as the AOS service PackagesLocalDirectory\bin
+        
+    .PARAMETER MetaDataDir
+        The path to the meta data directory for the environment
+        
+        Default path is the same as the aos service PackagesLocalDirectory
+        
+    .PARAMETER Module
+        Name of the Module to analyse
+        
+    .PARAMETER Model
+        Name of the Model to analyse
+        
+    .PARAMETER XmlLog
+        Path where you want to store the Xml log output generated from the best practice analyser
+
+	.PARAMETER PackagesRoot
+        Indicates to use binary metadata
+        
+    .EXAMPLE
+        PS C:\> Invoke-D365BestPractice -module "ApplicationSuite" -model "MyOverLayerModel"
+        
+        This will execute the best practice checks against MyOverLayerModel in the ApplicationSuite Module
+        
+    .NOTES
+        Author: Gert Van Der Heyden (@gertvdheyden)
+        
+#>
+
+function Invoke-D365BestPractice {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseShouldProcessForStateChangingFunctions', '')]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false, Position = 1 )]
+        [string] $BinDir = "$Script:PackageDirectory\bin",
+
+        [Parameter(Mandatory = $false, Position = 2 )]
+        [string] $MetaDataDir = "$Script:MetaDataDir",
+
+        [Parameter(Mandatory = $true, Position = 3 )]
+        [Alias('Package')]
+        [string] $Module,
+
+        [Parameter(Mandatory = $true, Position = 4 )]
+        [string] $Model,
+
+        [Parameter(Mandatory = $false, Position = 5 )]
+        [string] $XmlLog = (Join-Path $Script:DefaultTempPath "BPCheckLogcd.xml"),
+
+		[Parameter(Mandatory = $false, Position = 6 )]
+        [switch] $PackagesRoot
+
+
+    )
+    
+    if (-not (Test-PathExists -Path $MetaDataDir, $BinDir -Type Container)) {return}
+
+    $executable = Join-Path $BinDir "xppbp.exe"
+    if (-not (Test-PathExists -Path $executable -Type Leaf)) {return}
+
+    $param = @(
+        "-metadata=`"$MetaDataDir`"",
+        "-all",
+        "-module=`"$Module`"",
+        "-model=`"$Model`"",
+        "-xmlLog=`"$XmlLog`""
+        )
+	
+	if ($PackagesRoot -eq $true)
+	{
+		$param +="-packagesroot=`"$MetaDataDir`""
+	}
+
+    Write-PSFMessage -Level Verbose -Message "Starting the $executable with the parameter options." -Target $param
+    Start-Process -FilePath $executable -ArgumentList  ($param -join " ") -NoNewWindow -Wait
```

Comment:
We just implemented a new centralized solution to execute / invoke external executables. If you believe that you are up for the task to refactor it - cool, if not, let me know and I'll be happy to rewrite it to match the new way of doing it.

Inspiration can be found here:
https://github.com/d365collaborative/d365fo.tools/blob/9168515983cacbdc86593660c773c760eb2f540a/d365fo.tools/functions/invoke-d365modulecompile.ps1#L101-L110